operator/pkg/apis/infrastructure/v1alpha1/controlplane.go

```diff
@@ -41,47 +42,39 @@ type ControlPlaneList struct {
 // master and etcd are configured to run. By default, KIT uses all the default
 // values and ControlPlaneSpec can be empty.
 type ControlPlaneSpec struct {
-	Infrastructure    `json:",inline"`
 	KubernetesVersion string     `json:"kubernetesVersion,omitempty"`
 	Master            MasterSpec `json:"master,omitempty"`
 	Etcd              ETCDSpec   `json:"etcd,omitempty"`
 }
 
-type Infrastructure struct {
-	VPCCidr        string   `json:"vpccidr,omitempty"`
-	Zones          []string `json:"zones,omitempty"`
-	PrivateSubnets []string `json:"privateSubnets,omitempty"`
-}
-
 // MasterSpec provides a way for the user to configure master instances and
 // custom flags for components running on master nodes like apiserver, KCM and
 // scheduler.
 type MasterSpec struct {
 	Instances         `json:",inline"`
-	Scheduler         *Config `json:"scheduler,omitempty"`
-	ControllerManager *Config `json:"controllerManager,omitempty"`
-	APIServer         *Config `json:"apiServer,omitempty"`
+	Scheduler         *Deployment `json:"scheduler,omitempty"`
+	ControllerManager *Deployment `json:"controllerManager,omitempty"`
+	APIServer         *Deployment `json:"apiServer,omitempty"`
 }
 
 // ETCDSpec provides a way to configure the etcd nodes and args which are passed to the etcd process.
 type ETCDSpec struct {
-	Instances `json:",inline"`
-	*Config   `json:",inline"`
+	Instances   `json:",inline"`
+	*Deployment `json:",inline"`
 }
 
-// Config provides a generic way to pass in args and images to master and etcd
+// Deployment provides a generic way to pass in args and images to master and etcd
 // components. If a user wants to change the QPS they need to provide the
 // following flag with the desired value -`kube-api-qps:100` in the args.
-type Config struct {
-	Args  map[string]string `json:"args,omitempty"`
-	Image string            `json:"image,omitempty"`
+type Deployment struct {
+	Replicas int         `json:"replicas,omitempty"`
+	Spec     *v1.PodSpec `json:"spec,omitempty"`
 }
 
 // Instances denotes how the infrastructure of a particular components looks
 // like, if a user wants to use a specific AMI ID, they can provide this in the
 // Instances for the corresponding component.
 type Instances struct {
-	AMI   string `json:"ami,omitempty"`
-	Type  string `json:"type,omitempty"`
-	Count int    `json:"count,omitempty"`
+	AMI  string `json:"ami,omitempty"`
```

Comment:
I'm not sure AMI is enough. You'll likely need to control user data as well. If we're using Karpenter, can we do this with NodeSelectors to customer LaunchTemplates (w/ custom ami)? That will allow us to drive everything via the corev1.DeploymentSpec

Reply:
What do you need custom AMI for in the control plane, any examples?

For any example you provide, I can think of a use case for user data. If you're going with custom AMI support, I think you need custom LT support.

Reply:
We could potentially folow the same model as elsewhere and default the userdata to whatever the EKS Control Plane AMI expects.

Reply:
We can always add this later if needed. For now I am just keeping this limited.